USERMANUAL.md

```diff
@@ -0,0 +1,596 @@
+    row_filter_id: NONE
+    rule_ids:
+      - UNIQUE:
+          dummy: 0 # Due to bug, a CUSTOM_SQL_STATEMENT requires at least one parameter. Will be fixed in the next version.
+```
+
+Full example: [Rule](docs/examples/aggregate_conditions.yaml#38) and [rule binding](docs/examples/aggregate_conditions.yaml#47)
+
+
+#### Unique Composed Value Constraint (over multiple columns)
+
+Ensures that the combination of values from several columns are all unique across the table.
+
+Note that the row count that is reported for this rule will be the number of elements that are duplicated. If one element is duplicated three times, the row count returned will be 1.
+
+```yaml
+rules:
+  UNIQUE_MULTIPLE:
+    rule_type: CUSTOM_SQL_STATEMENT
+    custom_sql_arguments:
+      - columns
+    params:
+      custom_sql_statement: |-
+        select $columns from data
+        group by $columns
+        having count(*) > 1
+        
+rule_bindings:
+  UNIQUENESS_MULTIPLE:
+    entity_id: TEST_DATA
+    column_id: UNIQUE_KEY
+    row_filter_id: NONE
+    rule_ids:
+      - UNIQUE_MULTIPLE:
+          columns: unique_key, date
+```
+
+Full example: [Rule](docs/examples/aggregate_conditions.yaml#38) and [rule binding](docs/examples/aggregate_conditions.yaml#47)
+
+
+
+#### Maximum Gap
+
+Do not allow the gap between two subsequent values to be greater than a threshold.
+
+In this example, “two subsequent values” are two rows that are subsequent when sorting by the column. This works well for example when the column is a message timestamp, and we don’t want the time between messages to be too big. The example also assumes that the column being inspected is of type `TIMESTAMP`, wich requires the function `DATETIME_DIFF` to calculate the interval.
+
+But if we have a value, for example, current, at different points in time, and we don’t want the current to jump more than a threshold value between two measurements, we need to change the order by clause in the custom_sql_statement, to order by the time column.
+
+```yaml
+rules:
+  GAP:
+    rule_type: CUSTOM_SQL_STATEMENT
+    custom_sql_arguments:
+      max_gap
+    params:
+      custom_sql_statement: |-
+        with lagged_data as (
+          select $column, lag($column) over (order by $column ASC) as prev 
+          from data
+        )
+        select $column from lagged_data 
+        where prev is not null and datetime_diff($column, prev, HOUR) > $max_gap
+        
+rule_bindings:
+    entity_id: TEST_DATA
+    column_id: TIMESTAMP
+    row_filter_id: NONE
+    rule_ids:
+      - GAP:
+          max_gap: 24
+```
+
+Full example: [Rule](docs/examples/aggregate_conditions.yaml#38) and [rule binding](docs/examples/aggregate_conditions.yaml#47)
+
+
+
+#### Outliers by z-score
+
+The z-score is the deviation from the mean in units of the standard deviation. If the values in a column have a mean of 10 and  a standard deviation of 2, then a value of 16 in this column has a z-score of (10-16)/2 = 3.
```

Comment:
love it :) I suggest adding a unit to 3 to "3 standard deviations"

Reply:
Yes. Also note that `(10-16)/2 != 3` XD